File: livingrimoire start here/LivinGrimoire java/LivinGrimoire/Neuron.java
package LivinGrimoire;

import java.util.ArrayList;
import java.util.Hashtable;

// used to transport algorithms to other classes
public class Neuron {
    private Hashtable<Integer,ArrayList<Algorithm>> defcons = new Hashtable<>();

    public Neuron() {
        for (int i = 1; i < 6; i++) {
            defcons.put(i,new ArrayList<Algorithm>());
        }
    }
    public void insertAlg(int priority, Algorithm alg){
        if ((0< priority) && (priority < 6)){
            if (defcons.get(priority).size() < 4){
                defcons.get(priority).add(alg);
            }
        }
    }
    public Algorithm getAlg(int defcon){
        if (defcons.get(defcon).size() > 0){
            Algorithm temp = defcons.get(defcon).get(0);
            defcons.get(defcon).remove(0);
            return temp;
        }
        return null;
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/livinGrimoire/Neuron.kt
package livinGrimoire

import java.util.*

// used to transport algorithms to other classes
class Neuron {
    private val defcons = Hashtable<Int, ArrayList<Algorithm>>()

    init {
        for (i in 1..5) {
            defcons[i] = ArrayList()
        }
    }

    fun insertAlg(priority: Int, alg: Algorithm) {
        if (priority in 1..5) {
            if (defcons[priority]!!.size < 4) {
                defcons[priority]!!.add(alg)
            }
        }
    }

    fun getAlg(defcon: Int): Algorithm? {
        if (defcons[defcon]!!.size > 0) {
            val temp = defcons[defcon]!![0]
            defcons[defcon]!!.removeAt(0)
            return temp
        }
        return null
    }
}